provider/ansible/gcp_utils.py

```diff
@@ -119,6 +119,30 @@ def patch(self, url, body=None, **kwargs):
         kwargs.update({'json': body})
         return self.full_patch(url, **kwargs)
 
+    def list(self, url, params=None, array_name='items',
+             pageToken='nextPageToken', callback=None, **kwargs):
+        """
+        This should be used for calling the GCP list APIs. It will return
+        an array of items
+
+        This takes a callback to a `return_if_object(module, response)`
+        function that will decode the response + return a dictionary. Some
+        modules handle the decode + error processing differently, so we should
+        defer to the module to handle this.
+        """
+        resp = callback(self.module, self.full_get(url, params, **kwargs))
```

Comment:
`callback` is an optional arg; since it's actually required a better method signature would be `list(self, url, callback, params=None, ...`